code/datums/components/lock_on_cursor.dm

```diff
@@ -0,0 +1,202 @@
+#define LOCKON_IGNORE_RESULT "ignore_my_result"
+#define LOCKON_RANGING_BREAK_CHECK if(current_ranging_id != this_id){return LOCKON_IGNORE_RESULT}
+
+/**
+ * ### Lock on Cursor component
+ *
+ * Finds the nearest targets to your cursor and passes them into a callback, also drawing an icon on top of them.
+ */
+/datum/component/lock_on_cursor
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	// Appearance to overlay onto whatever we are targetting
+	var/mutable_appearance/lock_appearance
+	/// Current images we are displaying to the client
+	var/list/image/lock_images
+	/// Typecache of things we are allowed to target
+	var/list/target_typecache
+	/// Cache of weakrefs to ignore targetting formatted as `list(weakref = TRUE)`
+	var/list/immune_weakrefs
+	/// Number of things we can target at once
+	var/lock_amount
+	/// Range to search for targets from the cursor position
+	var/lock_cursor_range
+	/// Weakrefs to current locked targets
+	var/list/locked_weakrefs
+	/// Callback to call when we have decided on our targets, is passed the list of final targets
+	var/datum/callback/on_lock
+	/// Callback to call in order to validate a potential target
+	var/datum/callback/can_target_callback
+	/// Full screen overlay which is used to track mouse position
+	var/atom/movable/screen/fullscreen/cursor_catcher/lock_on/mouse_tracker
+	/// Ranging ID for some kind of tick check safety calculation
+	var/current_ranging_id = 0
+
+/datum/component/lock_on_cursor/Initialize(range = 5, list/typecache = list(), amount = 1, list/immune = list(), icon = 'icons/mob/silicon/cameramob.dmi', icon_state = "marker", datum/callback/when_locked, datum/callback/target_callback)
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	if (amount < 1 || range < 0)
+		CRASH("Invalid range or amount argument")
+	lock_cursor_range = range
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar

Use the same variable names for stuff like this.
i.e.
```
src.lock_cursor_range = lock_cursor_range
```
etc for all variables that apply in this proc